.github/workflows/ci.yml

```diff
@@ -122,33 +122,27 @@ jobs:
       if: runner.os == 'macOS'
       run: brew install boost
 
-    - name: Update CMake
-      uses: jwlawson/actions-setup-cmake@v2.0
-
-    - name: Cache wheels
-      if: runner.os == 'macOS'
-      uses: actions/cache@v4
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
       with:
-        # This path is specific to macOS - we really only need it for PyPy NumPy wheels
-        # See https://github.com/actions/cache/blob/master/examples.md#python---pip
-        # for ways to do this more generally
-        path: ~/Library/Caches/pip
-        # Look to see if there is a cache hit for the corresponding requirements file
-        key: ${{ runner.os }}-pip-${{ matrix.python }}-x64-${{ hashFiles('tests/requirements.txt') }}
+        enable-cache: true
 
     - name: Prepare env
-      run: |
-        python -m pip install -r tests/requirements.txt
+      run: uv pip install --python=python --system -r tests/requirements.txt
 
     - name: Setup annotations on Linux
       if: runner.os == 'Linux'
-      run: python -m pip install pytest-github-actions-annotate-failures
+      run: uv pip install --python=python --system pytest-github-actions-annotate-failures
+
+    - name: Use Ninja except on Windows
```

Comment:
A terse comment to explain why not Windows would be nice. (E.g. I have no idea, does ninja not work on Windows at all, or is there a glitch?)

Reply:
"Windows Ninja is broken and I don't have time to figure it out"? :) 

(It's the can't import the shared objects issue we've seen before on 3.8+)

Reply:
`# TODO Resolve Windows Ninja shared object issue on Python 3.8+`
?